lerobot/common/robot_devices/motors/dynamixel.py

```diff
@@ -0,0 +1,588 @@
+    """
+    Convert from motor position in [-2048, 2048] to radian in [-pi, pi]
+    """
+    return (position / 2048) * 3.14
+
+
+def motor_angle_to_position(angle: np.ndarray) -> np.ndarray:
+    """
+    Convert from radian in [-pi, pi] to motor position in [-2048, 2048]
+    """
+    return ((angle / 3.14) * 2048).astype(np.int64)
+
+
+# def pwm2vel(pwm: np.ndarray) -> np.ndarray:
+#     """
+#     :param pwm: numpy array of pwm/s joint velocities
+#     :return: numpy array of rad/s joint velocities
+#     """
+#     return pwm * 3.14 / 2048
+
+
+# def vel2pwm(vel: np.ndarray) -> np.ndarray:
+#     """
+#     :param vel: numpy array of rad/s joint velocities
+#     :return: numpy array of pwm/s joint velocities
+#     """
+#     return (vel * 2048 / 3.14).astype(np.int64)
+
+
+def get_group_sync_key(data_name, motor_names):
+    group_key = f"{data_name}_" + "_".join(motor_names)
+    return group_key
+
+def get_result_name(fn_name, data_name, motor_names):
+    group_key = get_group_sync_key(data_name, motor_names)
+    rslt_name = f"{fn_name}_{group_key}"
+    return rslt_name
+
+def get_queue_name(fn_name, data_name, motor_names):
+    group_key = get_group_sync_key(data_name, motor_names)
+    queue_name = f"{fn_name}_{group_key}"
+    return queue_name
+
+def get_log_name(var_name, fn_name, data_name, motor_names):
+    group_key = get_group_sync_key(data_name, motor_names)
+    log_name = f"{var_name}_{fn_name}_{group_key}"
+    return log_name
+
+class TorqueMode(enum.Enum):
+    ENABLED = 1
+    DISABLED = 0
+
+
+class OperatingMode(enum.Enum):
+    VELOCITY = 1
+    POSITION = 3
+    EXTENDED_POSITION = 4
+    CURRENT_CONTROLLED_POSITION = 5
+    PWM = 16
+    UNKNOWN = -1
+
+
+class DriveMode(enum.Enum):
+    NON_INVERTED = 0
+    INVERTED = 1
+
+
+class DynamixelMotorsBus:
+    """
+    Example of usage for 1 motor connected to the bus:
+    ```python
+    motor_name = "gripper"
+    motor_index = 6
+    motor_model = "xl330-m077"
+
+    robot = DynamixelMotorsBus(
+        port="/dev/tty.usbmodem575E0031751",
+        motors={motor_name: (motor_index, motor_model)},
+    )
+    robot.connect()
+
+    while True:
+        robot.teleop_step()
+    ```
+    """
+    def __init__(
+        self,
+        port: str,
+        motors: dict[str, tuple[int, str]],
+        extra_model_control_table: dict[str, list[tuple]] | None = None,
+    ):
+        self.port = port
+        self.motors = motors
+
+        self.model_ctrl_table = deepcopy(MODEL_CONTROL_TABLE)
+        if extra_model_control_table:
+            self.model_ctrl_table.update(extra_model_control_table)
+
+        self.port_handler = None
+        self.packet_handler = None
+        self.calibration = None
+        self.is_connected = False
+        self.group_readers = {}
+        self.group_writers = {}
+        self.logs = {}
+
+    def connect(self):
+        if self.is_connected:
+            raise ValueError(f"KochRobot is already connected.")
```

Comment:
Nit: Robot or MotorBus is already connected instead ?

Reply:
Done